artemis-server/src/main/java/org/apache/activemq/artemis/core/server/protocol/websocket/WebSocketFrameEncoder.java

```diff
@@ -63,5 +63,7 @@ private void writeContinuationFrame(ChannelHandlerContext ctx, ByteBuf byteBuf,
          byteBuf.readBytes(fragment, length);
          ctx.writeAndFlush(new ContinuationWebSocketFrame(finalFragment, 0, fragment), promise);
       }
+
+      byteBuf.release();

```

Comment:
would be possible to add a test?

Reply:
Yes I will add a test, unit test should be easy anyway I am trying to understand how to reproduce it in a consistent way because connecting and desconnecting does not trigger the issue at least inmac, assuming due to kqueue vs the linux impl. I will try to dig more if you have an idea about a good test I can try to create it

Reply:
@ldebello a mock test perhaps?

Reply:
I have added a test using a direct buffer to verify the release is called. the test is using a direct buffer and I checked the increment and decrement is called I was not able to add that condition in the test because netty does not expose that information

Reply:
is the test good enough?